### PR TITLE
ENH add implementation for init and push

### PR DESCRIPTION
**Pull request by adrinjalali**
Add implementation for `init` and `push`.

### CODE REVIEW

**Comment by adrinjalali** [comment]
@merveenoyan this should be ready for a review :)

File: skops/hf_hub/_hf_hub.py [hunk 1]
```diff
@@ -94,6 +169,15 @@ def push(*, repo_id: str, source: Union[str, Path], token: str = None):
...
         logged in using ``huggingface-cli login``.
 
+    commit_message: str, optional
+        The commit message to be used when pushing to the repo.
+
+    create_remote: bool, optional
+        Whether to create the remote repository if it doesn't exist. If the
+        remote repository doesn't exist and this parameter is ``False``, it
+        raises an error. Otherwise it checks if the remote repository exists,
+        and would create it if it doesn't.
```
Comment (merveenoyan) [inline skops/hf_hub/_hf_hub.py:179]: If this exists, why do we create_repo in examples and tests?

File: skops/hf_hub/_hf_hub.py [hunk 2]
```diff
@@ -103,4 +187,23 @@ def push(*, repo_id: str, source: Union[str, Path], token: str = None):
...
+        except HTTPError:
+            client.create_repo(repo_id=repo_id, token=token, repo_type="model")
+
+    client.upload_folder(
+        repo_id=repo_id,
+        path_in_repo=".",
+        folder_path=source,
+        commit_message=commit_message,
+        commit_description=None,
+        token=token,
```
Comment (merveenoyan) [inline skops/hf_hub/_hf_hub.py:205]: Why is this needed?

File: skops/hf_hub/_hf_hub.py [hunk 3]
```diff
@@ -103,4 +187,23 @@ def push(*, repo_id: str, source: Union[str, Path], token: str = None):
...
+            client.model_info(repo_id=repo_id, token=token)
+        except HTTPError:
+            client.create_repo(repo_id=repo_id, token=token, repo_type="model")
+
+    client.upload_folder(
+        repo_id=repo_id,
+        path_in_repo=".",
+        folder_path=source,
+        commit_message=commit_message,
+        commit_description=None,
```
Comment (merveenoyan) [inline skops/hf_hub/_hf_hub.py:204]: Why is this needed? 😅

File: skops/hf_hub/_hf_hub.py [hunk 4]
```diff
@@ -103,4 +187,23 @@ def push(*, repo_id: str, source: Union[str, Path], token: str = None):
...
+        token=token,
+        repo_type=None,
```
Comment (merveenoyan) [inline skops/hf_hub/_hf_hub.py:206]: This is by default None and None refers to model BTW

Comment (adrinjalali) [reply skops/hf_hub/_hf_hub.py:206]: Yes, same as above, I'm just explicitly setting it in case in the future the default value changes.

File: skops/hf_hub/_hf_hub.py (hunk 1 above)
Comment (adrinjalali) [reply skops/hf_hub/_hf_hub.py:179]: I've removed it from the example, but the test needs to test different scenarios, and the user might create the repo before calling this method, so the test makes sure that case is tested as well.

File: skops/hf_hub/_hf_hub.py (hunk 4 above)
Comment (merveenoyan) [reply skops/hf_hub/_hf_hub.py:206]: ah gotcha!

**Review by merveenoyan — APPROVED** [review]
I didn't catch anything else other than above comments, LGTM
I had a couple of concerns around requirements, like maybe users might want to set the versions to be between X and Y but it's a big brain comment for early stage 😅
